mesonbuild/dependencies/ui.py

```diff
@@ -45,6 +45,11 @@ def __init__(self, name: str, environment: 'Environment', kwargs: T.Dict[str, T.
             self.link_args = ['-framework', 'OpenGL']
             # FIXME: Detect version using self.clib_compiler
             return
+        if self.env.machines[self.for_machine].is_linux():
+            lib = self.clib_compiler.find_library('GL', environment, [])
+            if lib:
+                self.is_found = True
+                self.link_args = lib
```

Comment:
I suppose we should also check via `self.clib_compiler.has_header('GL/gl.h', '', env)[0]` if the header is installed, in case only the runtime library is available?

Reply:
I've accounted for this scenario, and if only the runtime library is available then I throw a specific exception in 8f9967d.
Otherwise I imagine trying to hunt down why something went wrong is going to get rather frustrating. I hope this wouldn't mess with anything else - my reasoning was that the compilation will fail without GL anyway. However, if `required: false` was set, I hope this would catch and then ignore any such raised exception?

Reply:
It doesn't catch and ignore any such raised exception.

It *does* however specifically catch DependencyException, that's why we have that special derived exception class, so you did that correctly. :)

...

We don't usually bother to raise such an exception, anecdotally, but doing so seems fine so sure, why not. :)